config/opal_check_libnl_version.m4

```diff
@@ -0,0 +1,50 @@
+dnl -*- shell-script -*-
+dnl
+dnl Copyright (c) 2015      Research Organization for Information Science
+dnl                         and Technology (RIST). All rights reserved.
+dnl $COPYRIGHT$
+dnl
+dnl Additional copyrights may follow
+dnl
+dnl $HEADER$
+dnl
+
+dnl OPAL_CHECK_LIBNL_INIT()
+dnl --------------------------------------------------------------------
+AC_DEFUN([OPAL_CHECK_LIBNL_INIT], [
+    opal_libnl_version=0
+    opal_libnl_libs=""
+    opal_libnl3_libs=""
+])
+
+dnl OPAL_CHECK_LIBNL_VERSION(lib)
+dnl --------------------------------------------------------------------
+AC_DEFUN([OPAL_CHECK_LIBNL_VERSION], [
+     AC_LANG_PUSH(C)
+     rm -f conftest_c.$ac_ext
+      cat > conftest_c.$ac_ext << EOF
+int main(int argc, char *argv[]) {
+    return 0;
+}
+EOF
+     OPAL_LOG_COMMAND([$CC -o conftest $CFLAGS $CPPFLAGS conftest_c.$ac_ext $LDFLAGS -l$1],
```

Comment:
It would be good to avoid OPAL_LOG_COMMAND with explicitly compiling a source and then running ldd on it (e.g., ldd doesn't exist on OS X.  I know libnl isn't on OS X, but we should probably still avoid OS-specific constructs in configure when possible).

It might be better to AC_LINK_IFELSE with symbols that are known to be in libnl vs. libnl3 to determine which libnl you have.  That would seem to be more portable and resilient if the output of ldd varies, or the exact names of the libraries varies between different Linux distros.